File: JavaScript/539-Minimum Time Difference.js
/**
 * @param {string[]} timePoints
 * @return {number}
 */
var findMinDifference = function(timePoints) {
    let time = [];
    for (let timePoint of timePoints){
        time.push(60*parseInt(timePoint.substr(0,2)) + parseInt(timePoint.substr(3,5)));
    }
    
    time.sort(function(a,b){
        return a-b;
    });
    
    let res = 1440;
    for (let i=1; i<time.length; i++){
        res = Math.min(res, time[i] - time[i-1]);
    }
    
    res = Math.min(res, 1440 - (time[time.length-1] - time[0]) );
    
    return res;
};

File: C++/539-Minimum Time Difference.cpp
class Solution {
public:
    int findMinDifference(vector<string>& timePoints) {
        vector<int> arr;
        for (auto &t:timePoints) {
            int hour = stoi(t.substr(0, 2));
            int minute = stoi(t.substr(3, 2));
            arr.push_back(hour*60+minute);
        }
        sort(arr.begin(), arr.end());
        int res = INT_MAX;
        for (int i=1; i<arr.size(); i++) {
            res = min(res, arr[i]-arr[i-1]);
            if (res == 0) return 0;
        }
        res = min(res, 1440-(arr[arr.size()-1]-arr[0]) );

        return res;
    }
};